tools/integration_tests/read_cache/read_only_test.go

```diff
@@ -0,0 +1,128 @@
+package read_cache
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/client"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/log_parser/json_parser/read_logs"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/test_setup"
+)
+
+const (
+	MiB          = 1024 * 1024
+	fileSize     = 3 * MiB
+	chunksRead   = fileSize / MiB
+	testFileName = "foo"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type testStruct struct {
+	flags         []string
+	storageClient *storage.Client
+	ctx           context.Context
+}
+
+func (s *testStruct) Setup(t *testing.T) {
+	if setup.MountedDirectory() != "" {
+		t.Log("Skipping setup as tests are running test in GKE environment")
+		return
+	}
+	// Mount GCSFuse.
+	if err := mountFunc(s.flags); err != nil {
+		t.Errorf("Failed to mount GCSFuse: %v", err)
+	}
+	setup.SetMntDir(mountDir)
+	testDirPath = client.SetupTestDirectoryForROMount(s.ctx, s.storageClient, testDirName)
```

Comment:
Without `--ro` mount also testDir and testFile creating by this way?

Reply:
Yes, for both we are using go storage client. I will rename the method.